### DISCUSSION TITLE
Declare nDim domains and solve nDim PDEs

### DISCUSSION THREAD

**Discussion by Marti2405** [lululxvi/deepxde#1399] [Q&A]
Hello,
I have a problem when increasing the number of dimensions of my PDE resolutions.

Lets take a 4D spatial domain. Here is how I declare it:

        xmin = [x_min1, x_min2, x_min3, x_min4]  # Lower bounds in 4D spatial domain
        xmax = [x_max1, x_max2, x_max3, x_max4]  # Upper bounds in 4D spatial domain
        
        # Spatial Domain
        geom = dde.geometry.geometry_nd.Hypercube(xmin,xmax)

The error is raised in my pde function when I try to acces the i=2,j=2 element of the hessian. I printed the size of my input and realized that it is 2 dimensional, this is why my hessian is not a 4x4 matrix. Here is the pde function:
        
        def pde(X,u): 
            print("X shape:", X.shape)  # -> this returns (None, 2)
            du_xx= dde.grad.hessian(u, X, i=0, j=0) 
            du_yy = dde.grad.hessian(u, X, i=1, j=1)
            du_zz = dde.grad.hessian(u, X, i=2, j=2)
            du_pp = dde.grad.hessian(u,X ,i=3,j=3)
            return du_yy + du_xx + du_zz + du_pp -1

I don't understand where the problem comes from. Is it when I declare the spatial domain? Or maybe the deepxde.data.PDE function doesn't accept high dimensional spatial domains? 

Does someone know how to fix this? Thank you for your time.

**Comment by lululxvi** [comment]
This is strange. It should work.
